bika/lims/content/duplicateanalysis.py

```diff
@@ -89,8 +101,13 @@ def getSiblings(self):
             if IRequestAnalysis.providedBy(analysis):
                 # We exclude here all analyses that do not have an analysis
                 # request associated (e.g. IReferenceAnalysis)
-                if analysis.getRequestUID() == requestuid:
-                    siblings.append(analysis)
+                if analysis.getRequestUID() != requestuid:
+                    continue
+
+                if retracted == False and in_state(analysis, retracted_states):
```

Comment:
better use `if retracted is False and in_state(analysis, retracted_states)` (compare bool with `is`)